crates/astria-bridge-withdrawer/src/bridge_withdrawer/submitter/mod.rs

```diff
@@ -262,21 +265,24 @@ async fn submit_tx(
                 async move {}
             },
         );
-    let res = tryhard::retry_fn(|| {
+    let check_tx = tryhard::retry_fn(|| {
         let client = client.clone();
         let tx = tx.clone();
         let span = info_span!(parent: span.clone(), "attempt send");
-        async move { client.submit_transaction_commit(tx).await }.instrument(span)
+        async move { client.submit_transaction_sync(tx).await }.instrument(span)
     })
     .with_config(retry_config)
     .await
     .wrap_err("failed sending transaction after 1024 attempts");
 
-    state.set_sequencer_connected(res.is_ok());
+    state.set_sequencer_connected(check_tx.is_ok());
 
     metrics.record_sequencer_submission_latency(start.elapsed());
 
-    res
+    let check_tx = check_tx?;
```

Comment:
check the `check_tx` code here, ie move this block here: `if let tendermint::abci::Code::Err(check_tx_code) = check_tx.code`

if `check_tx` returns an error code, then the next call to `wait_for_tx_inclusion` can never be successful.